### PR TITLE
Refactoring and improving code quality for network.py, channel.py and user.py

### PR DESCRIPTION
**Pull request by krishnacharya**
## What? Why?
Fix # . (Add Reference to the issue if any)

Changes proposed in this pull request:
-adds extended_stop_words, build centroid, get_nick_receiver and iter_nicks in user.py
-adds adds build_conversation,conv_helper,build_conv_csv,conv_append,build_mean_list,res_helper, get_nick_sen_rec and conv_mat_diff to channel.py
-adds get_top_indices, get_indices_to_delete, add_common_users_weighted_edge and add_channel_weighted_edge to network.py
-adds splice_find, get_nick_sen_rec to util.py

## Checks
- [x] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Make sure you have added required documentation
- [x] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Any images?
Message Number graph for a single day(2nd Jan) earlier
mng2
after:
mng2

Nick Change graph before:
ncg11

Nick Change graph after:
ncg11

Message Time graph:
before : 
mtg3

after:
mtg3

channel.py
Before:
conv_len:
[conv_len.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839163/conv_len.xlsx)

conv_ref_time:
[conv_ref_time.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839165/conv_ref_time.xlsx)

resp_time:
[resp_time.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839166/resp_time.xlsx)

After:
conv_len:
[conv_len.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839178/conv_len.xlsx)
conv_ref_time:
[conv_ref_time.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839180/conv_ref_time.xlsx)

resp_time:
[resp_time.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839182/resp_time.xlsx)

keywords before:
[op.txt](https://github.com/prasadtalasila/IRCLogParser/files/839185/op.txt)

keywords after:
[op10.txt](https://github.com/prasadtalasila/IRCLogParser/files/839186/op10.txt)

## Notify reviewers
@rohangoel96

### CODE REVIEW

**Comment by krishnacharya** [comment]
channel.py has a C grade because of its cyclomatic complexity, i have fixed all of its code duplication though.
network.py has a F grade because of some loops requiring us call a graphs out degree, in degree etc refactoring this seems ineffective as we will rewrite code for this thrice even in the refactored function, but im working on a dictionary approach to solve this.
user.py has been brought upto an A

**Comment by krishnacharya** [comment]
The dictionary implementation ie using degree maps has done the trick and now network.py is now at A.
degree analysis_on_graph before : 
in:
[in_degree.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839405/in_degree.xlsx)
out:
[out_degree.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839406/out_degree.xlsx)
total:
[total_degree.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839407/total_degree.xlsx)

After
in:
[in_degree.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839409/in_degree.xlsx)
out:
[out_degree.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839410/out_degree.xlsx)
total:
[total_degree.xlsx](https://github.com/prasadtalasila/IRCLogParser/files/839411/total_degree.xlsx)

**Comment by rohangoel96** [comment]
@krishnacharya , kindly rectify the errors in travis build (https://travis-ci.org/prasadtalasila/IRCLogParser/jobs/211670051)

File: lib/analysis/network.py [hunk 1]
```diff
@@ -387,17 +421,26 @@ def give_userlist_where_degree_helper(degree_dict, degree):
...
+         """
+            Args:
+                nodes(List) : nodes_with_OUT/IN/TOTAL degree\
+                raw(List) : raw_in/out/total
+                degree_type(str) : "in" "out" or "all" , basically keys of degree_map          
+            returns:
+                raw(List)
+                nodes(List)
+         """  
+        for i in range(1, len(nodes)):
```
Comment (rohangoel96) [inline lib/analysis/network.py]: @krishnacharya  this is the line where the error is

**Pull request merged**